tests/test_maintel_apply_dof.py

```diff
@@ -0,0 +1,101 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import random
+import unittest
+
+import numpy as np
+from lsst.ts import standardscripts
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+from lsst.ts.observatory.control.utils.enums import DOFName
+from lsst.ts.standardscripts.maintel import ApplyDOF
+
+random.seed(47)  # for set_random_lsst_dds_partition_prefix
+
+logging.basicConfig()
+
+
+class TestApplyDOF(
+    standardscripts.BaseScriptTestCase, unittest.IsolatedAsyncioTestCase
+):
+    async def basic_make_script(self, index):
+        self.script = ApplyDOF(index=index)
+
+        # Mock the MTCS
+        self.script.mtcs = MTCS(
+            domain=self.script.domain,
+            intended_usage=MTCSUsages.DryTest,
+            log=self.script.log,
+        )
+        self.mtcs.mtaos.cmd_offsetDOF.DataType = unittest.mock.PropertyMock(
+            return_value=np.zeros(50).dtype
+        )
+        self.script.mtcs.mtaos.cmd_offsetDOF.start = self.apply_dofs
+        self.script.mtcs.assert_all_enabled = unittest.mock.AsyncMock()
+        self.script.mtcs.start_task = unittest.mock.AsyncMock()
+
+        self.state_0 = np.zeros(50)
+
+        return (self.script,)
+
+    async def apply_dofs(self, offset):
+        await asyncio.sleep(0.5)
+        self.state_0 += offset
+
+    async def test_configure(self) -> None:
+        # Try configure with minimum set of parameters declared
+        async with self.make_script():
+            config_dofs = {"M2_dz": 0.2, "Cam_dy": 0.3, "M1M3_B1": 0.5, "M2_B14": 0.7}
+
+            await self.configure_script(**config_dofs)
+
+            dofs = np.zeros(len(DOFName))
+            for key, value in config_dofs.items():
+                dofs[getattr(DOFName, key)] = value
+
+            assert all(self.script.dofs == dofs)
+
+    async def test_run(self) -> None:
+        # Start the test itself
+        async with self.make_script():
+            config_dofs = {"M2_dz": 0.2, "Cam_dy": 0.3, "M1M3_B1": 0.5, "M2_B14": 0.7}
+
+            await self.configure_script(**config_dofs)
+
+            dofs = np.zeros(len(DOFName))
+            for key, value in config_dofs.items():
+                dofs[getattr(DOFName, key)] = value
+
+            # Run the script
+            await self.run_script()
+
```

Comment:
Maybe add assertions about the things that you call and await that are mocks? e.g.,

```
self.script.mtcs.mtaos.cmd_offsetDOF.DataType.assert_called()
self.script.mtcs.rem.mtaos.cmd_offsetDOF.start.assert_awaited_once()
```